inventory/environ.py

```diff
@@ -140,6 +136,10 @@ def getIndexerClustering(vars_scope):
     idxc_vars = vars_scope["splunk"]["idxc"]
     idxc_vars["label"] = os.environ.get("SPLUNK_IDXC_LABEL", idxc_vars.get("label"))
     idxc_vars["secret"] = os.environ.get("SPLUNK_IDXC_SECRET", idxc_vars.get("secret"))
+    # Rectify cluster replication + search factors
+    indexer_count = len(inventory.get("splunk_indexer", {}).get("hosts", []))
+    idxc_vars["replication_factor"] = min(indexer_count, int(idxc_vars.get("replication_factor", 0)))
```

Comment:
This may be a good spot to raise an exception to let users know the user-defined replication factor cannot be met instead of just correcting it for them?

Reply:
@nwang92 Yeah, I think this is the right spot for it